proxysql/build-binary-proxysql

```diff
@@ -12,7 +12,16 @@ set -o xtrace
 if [ -f /usr/bin/yum ]; then
     RHEL=$(rpm --eval %rhel)
     if [[ $RHEL -eq 8 ]]; then
-        sudo yum -y install cmake python2 gnutls-devel libtool || true
+        sudo yum -y install python2 gnutls-devel libtool || true
+        sudo yum remove -y cmake
+        sudo wget https://github.com/Kitware/CMake/releases/download/v3.19.8/cmake-3.19.8.tar.gz
+        sudo tar -zxvf cmake-3.19.8.tar.gz
+        cd cmake-3.19.8/
+        sudo ./bootstrap
+        sudo make
+        sudo make install
+        sudo ln -s /usr/local/bin/cmake /usr/bin/cmake
```

Comment:
Is there a specific reason for creating this softlink ?

Reply:
<img width="957" alt="Screenshot 2022-06-16 at 18 01 43" src="https://user-images.githubusercontent.com/25184101/174100109-1c03e077-4bca-48f4-8532-56ca46c8fb6e.png">

Reply:
Ah, now I understand why this behaviour is seen. You can retain the softlinks but let me explain the root cause.

Earlier, during the same shell session, cmake existed and that executable was found in /usr/bin, but that was uninstalled.

You then installed another cmake executable in /usr/local/bin.

The bash shell caches the first location that it finds for any external command that you use, which means it doesn't have to do an expensive search for the executable the next time you use the same command. The downside of this is that it won't bother to look again when you, later, install another executable by the same name, even if it's done in a directory that occurs earlier in $PATH than the original location.

The solution to this issue is to empty the cached locations of executables that bash keeps. This is done with 
(hash -r) . To only forget the location of the cmake executable, use hash -d cmake.